Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -0,0 +1,268 @@
+            callUpdateCallback();
+        });
+    }
+
+    // FIXME: Set rendering suppression for view transitions to false.
+
+    if (m_document->activeViewTransition() == this)
+        clearViewTransition();
+
+    m_phase = ViewTransitionPhase::Done;
+
+    if (reason.hasException())
+        m_ready.second->reject(reason.releaseException());
+    else {
+        m_ready.second->rejectWithCallback([&] (auto&) {
+            return reason.releaseReturnValue();
+        }, RejectAsHandled::Yes);
+    }
+
+    m_updateCallbackDone.first->whenSettled([this, protectedThis = Ref { *this }]() {
+        switch (m_updateCallbackDone.first->status()) {
+        case DOMPromise::Status::Fulfilled:
+            m_finished.second->resolve();
+            break;
+        case DOMPromise::Status::Rejected:
+            m_finished.second->rejectWithCallback([&] (auto&) {
+                return m_updateCallbackDone.first->result();
+            }, RejectAsHandled::Yes);
+            break;
+        case DOMPromise::Status::Pending:
+            ASSERT_NOT_REACHED();
+            break;
+        }
+    });
+}
+
+// https://drafts.csswg.org/css-view-transitions/#ViewTransition-skipTransition
+void ViewTransition::skipTransition()
+{
+    if (m_phase != ViewTransitionPhase::Done)
+        skipViewTransition(Exception { ExceptionCode::AbortError, "Skipping view transition because skipTransition() was called."_s });
+}
+
+// https://drafts.csswg.org/css-view-transitions/#call-dom-update-callback-algorithm
+void ViewTransition::callUpdateCallback()
+{
+    if (!m_document)
+        return;
+
+    ASSERT(m_phase == ViewTransitionPhase::Done || m_phase == ViewTransitionPhase::PendingCapture);
+
+    Ref protectedThis { *this };
+
+    RefPtr<DOMPromise> callbackPromise;
+    if (!m_updateCallback) {
+        auto promiseAndWrapper = createPromiseAndWrapper(*m_document);
+        promiseAndWrapper.second->resolve();
+        callbackPromise = WTFMove(promiseAndWrapper.first);
+    } else {
+        auto result = m_updateCallback->handleEvent();
+        callbackPromise = result.type() == CallbackResultType::Success ? result.releaseReturnValue() : nullptr;
+        if (!callbackPromise || callbackPromise->isSuspended()) {
+            auto promiseAndWrapper = createPromiseAndWrapper(*m_document);
+            // FIXME: First case should reject with `ExceptionCode::ExistingExceptionError`.
+            if (result.type() == CallbackResultType::ExceptionThrown)
+                promiseAndWrapper.second->reject(ExceptionCode::TypeError);
+            else
+                promiseAndWrapper.second->reject();
+            callbackPromise = WTFMove(promiseAndWrapper.first);
+        }
+    }
+
+    if (m_phase != ViewTransitionPhase::Done)
+        m_phase = ViewTransitionPhase::UpdateCallbackCalled;
+
+    callbackPromise->whenSettled([this, weakThis = WeakPtr { *this }, callbackPromise] () mutable {
+        if (!weakThis)
```

Comment:
You need to assigned to a RefPtr and then null check. Otherwise, nothing protects `this` during the execution of this lambda:
```
RefPtr protectedThis = weakThis.get();
if (!protectedThis)
    return;
``` 

I know I said "no protectors" but lambda where you capture a weakPtr are an exception, since the "caller" on the lambda is not protecting this.